### PR TITLE
Added a new exercise `reverse-string`

### PR DESCRIPTION
**Pull request by suraj-mandal**
Hello, 

Here I have created a pull request, where I have added the exercise reverse-string. This exercise was not added to gleam track in exercism before, so I have taken an oppurtunity to add it.

This exercism is of easy category. I am intending to add more exercises of same category before migrating towards adding medium difficulty exercises and finally hard exercises which is my ultimate goal.

Please review the **PR** and if you like it please push it, since this is my first PR.

### Note:

The `<>` operator in gleam was throwing syntax error. So as a workaround I had to replace some parts of `exercise_generator.gleam` with `string.concat` so that the project generation is successful. I have not committed this file and as a result no changes can be reflected there.

But I would sincerely want a way so that gleam considers `<>` to be `concatenation` operator and not just throw some `syntax error`.

### CODE REVIEW

**Comment by jiegillet** [comment]
Thank you. 
could you have another pass at the PR? I can tell that you ran the generator many times because the config file (and the .gitignore file) are duplicated many times. You also didn’t add any config or ran the configlet commands suggested by the generator. 

@lpil this exercise was not added on the Elixir track because it’s too easy, barely more than hello world. I don’t really mind, but what do you think about it?

**Comment by suraj-mandal** [comment]
Ok sure.. i am checking it..

**Comment by suraj-mandal** [comment]
Hi! @jiegillet I updated gleam to 0.26. Also I ran those configlet commands as you requested. Can you check once more if everything is okay?

File: config.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "highlightjs_language": "rust"
   },
   "test_runner": {
-    "average_run_time": 3.0
+    "average_run_time": 3
```
Comment (jiegillet) [inline config.json:19]: Can you revert this change? It’s a bug from the generator script, sorry.

File: config.json [hunk 2]
```diff
@@ -562,6 +562,14 @@
           "bool"
         ],
         "difficulty": 6
+      },
+      {
+        "slug": "reverse-string",
+        "name": "TODO",
+        "uuid": "6cf90987-bc60-4de9-8f68-343ca5039c3b",
+        "practices": [],
+        "prerequisites": [],
```
Comment (jiegillet) [inline config.json:571]: Please add the name (String Revert) and add what the exercise practices and requires. In this case, it’s “string” for both.

File: config.json [hunk 3]
```diff
@@ -562,6 +562,14 @@
...
+        "difficulty": 5
```
Comment (jiegillet) [inline config.json:572]: That’s gotta be a 1 :)

File: exercises/practice/reverse-string/.meta/config.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "authors": [],
```
Comment (jiegillet) [inline exercises/practice/reverse-string/.meta/config.json:2]: Add your GitHub handle here, you are the author.

File: exercises/practice/reverse-string/test/reverse_string_test.gleam [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import gleeunit
+import gleeunit/should
+import reverse_string
+
+// If property based testing tools are available, a good property to test is reversing a string twice: reverse(reverse(string)) == string
```
Comment (jiegillet) [inline exercises/practice/reverse-string/test/reverse_string_test.gleam:5]: You can remove the comment

File: exercises/practice/reverse-string/test/reverse_string_test.gleam [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+pub fn main() {
+  gleeunit.main()
+}
+
+pub fn an_empty_string_test() {
+  reverse_string.reverse(value: "")
```
Comment (jiegillet) [inline exercises/practice/reverse-string/test/reverse_string_test.gleam:12]: Let’s remove the “value:” for all tests, stub and example solution, there is no ambiguity anywhere

File: exercises/practice/reverse-string/test/reverse_string_test.gleam [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  reverse_string.reverse(value: "racecar")
+  |> should.equal("racecar")
+}
+
+pub fn an_even_sized_word_test() {
+  reverse_string.reverse(value: "drawer")
+  |> should.equal("reward")
+}
+
+pub fn sentence_test_with_no_alphabets() {
```
Comment (jiegillet) [inline exercises/practice/reverse-string/test/reverse_string_test.gleam:41]: The test names must end in _test

Comment (jiegillet) [reply exercises/practice/reverse-string/test/reverse_string_test.gleam:41]: Is this one in the canonical data? If not, let’s leave it out for the sake of maintainability

**Review by jiegillet — COMMENTED** [review]
Thank you for adding an exercise. 

I don’t have access to a computer until tomorrow evening, I’m reviewing on my phone so I hope I didn’t miss anything. there are a few things to clean up. 
im also not certain we want to add this exercise since it’s so simple to do in Gleam, so maybe you will want to wait until we hear from @lpil about that (or get started on another exercise haha :p)

File: exercises/practice/reverse-string/.meta/example.gleam [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import gleam/string
+
+pub fn reverse(value value: String) -> String {
+  string.reverse(value)
```
Comment (lpil) [inline exercises/practice/reverse-string/.meta/example.gleam:4]: I think it would be cool to implement this without using `string.reverse`.

Some options could be to use string patterns with the `<>` operator to match on an explicit set of characters (so wouldn't work for all unicode), to convert the string to a bit string and match on utf codepoints (so it may mangle some unicode strings, but I imagine most languages' example solutions will too), or to use the `string.pop_grapheme` function to get "characters" from the string.

**Review by lpil — COMMENTED** [review]
Thanks!

RE `<>` giving a syntax error, I suspect you're using an older version of Gleam which doesn't support this. Could you double check you're on 0.26.1 or 0.26.2? :pray:

File: exercises/practice/reverse-string/.meta/example.gleam (hunk 8 above)
Comment (suraj-mandal) [reply exercises/practice/reverse-string/.meta/example.gleam:4]: Hi @Ipil! Yes I am working on these issues. There are lots of changes to be made as can be seen. No wonder I did sweat on my first commit. Fixing ASAP

Comment (lpil) [reply exercises/practice/reverse-string/.meta/example.gleam:4]: fab, thank you!!

Comment (suraj-mandal) [reply exercises/practice/reverse-string/.meta/example.gleam:4]: A stack based solution would do I guess..

**Comment by suraj-mandal** [comment]
> Thanks!
> 
> RE `<>` giving a syntax error, I suspect you're using an older version of Gleam which doesn't support this. Could you double check you're on 0.26.1 or 0.26.2? 🙏

I used 0.22 before but now I am using 0.26.1. Thing is I was testing on windows so there the gleam version was 0.22 and was geting this problem. Switching to gleam on wsl helped resolve it. and now the operator works without any issues.

File: exercises/practice/reverse-string/test/reverse_string_test.gleam (hunk 7 above)
Comment (suraj-mandal) [reply exercises/practice/reverse-string/test/reverse_string_test.gleam]: It isn't there, I added this test on my own. Yeah it was not added

File: exercises/practice/reverse-string/.meta/example.gleam (hunk 8 above)
Comment (suraj-mandal) [reply exercises/practice/reverse-string/.meta/example.gleam]: Hello I am creating a new pull request. Since for this I was working with main branch which was creating hassles for me. I made changes to a new branch. I am creating the same pull request however changing the branch here.

**Comment by jiegillet** [comment]
Closing in favor of #176
